linux.nix

```diff
@@ -0,0 +1,124 @@
+{ stdenv, runCommand, writeText, writeScriptBin, fetchurl, fetchFromGitHub, openssl, electron,
+coreutils, utillinux, procps, cluster,
+rawapp, master_config, cardanoPkgs }:
+
+let
+  # closure size TODO list
+  # openssl depends on perl
+  # electron depends on cups, which depends on avahi
+  daedalus_frontend = writeScriptBin "daedalus" ''
+    #!${stdenv.shell}
+
+    test -z "$XDG_DATA_HOME" && { XDG_DATA_HOME="''${HOME}/.local/share"; }
+    export DAEDALUS_DIR="''${XDG_DATA_HOME}/Daedalus"
+
+    cd "''${DAEDALUS_DIR}/${cluster}/"
+
+    exec ${electron}/bin/electron ${rawapp}
+  '';
+  cardanoProgs = runCommand "cardano" {} ''
+    mkdir -pv $out/bin/
+    cp ${cardanoPkgs.cardano-sl-wallet}/bin/cardano-node $out/bin/
+    cp ${cardanoPkgs.cardano-sl-tools}/bin/cardano-launcher $out/bin/
+  '';
+  slimOpenssl = runCommand "openssl" {} ''
+    mkdir -pv $out/bin/
+    cp ${openssl}/bin/openssl $out/bin/
+  '';
+  configFiles = runCommand "cardano-config" {} ''
+    mkdir -pv $out
+    cd $out
+    cp -vi ${cardanoPkgs.cardano-sl.src + "/configuration.yaml"} configuration.yaml
+    cp -vi ${cardanoPkgs.cardano-sl.src + "/mainnet-genesis-dryrun-with-stakeholders.json"} mainnet-genesis-dryrun-with-stakeholders.json
+    cp -vi ${cardanoPkgs.cardano-sl.src + "/mainnet-genesis.json"} mainnet-genesis.json
+    cp -vi ${cardanoPkgs.cardano-sl.src + "/../log-configs/daedalus.yaml"} daedalus.yaml
+    cp -vi ${topologies.${cluster}} topology.yaml
+  '';
+  topologies = {
+    # TODO DEVOPS-690 integration
+    mainnet = writeText "topology.yaml" ''
+      wallet:
+        relays:
+          [
+            [
+              { host: relays.cardano-mainnet.iohk.io }
+            ]
+          ]
+        valency: 1
+        fallbacks: 7
+    '';
+    staging = writeText "topology.yaml" ''
+      wallet:
+        relays:
+          [
+            [
+              { host: relays.awstest.iohkdev.io }
+            ]
+          ]
+        valency: 1
+        fallbacks: 7
+    '';
+  };
+  perClusterConfig = {
+    mainnet = {
+      key = "mainnet_wallet_macos64";
+    };
+    staging = {
+      key = "mainnet_dryrun_wallet_macos64";
+    };
+  };
+  launcherConfig = writeText "launcher-config.json" (builtins.toJSON {
```

Comment:
Indeed, depends on DEVOPS-690 which is almost ready (at least the gist of it - configuration dhall-to-yaml)